### PR TITLE
Main PR

### PR DESCRIPTION
**Pull request by brennanmcmicking**
This is the big chungus PR with all of my changes since August. I haven't done a good job of merging often because nobody else has been working on the project.

### CODE REVIEW

File: src/components/courses.rs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+                    }
+                }
+            }
+        }
+    )
+}
+
+fn sections(schedule_id: &String, sections: Vec<&Section>, selected: &[u64]) -> Markup {
+    html!(
+        div class="flex flex-col gap-2 py-2 border-t" {
```
Comment (malcolmseyd) [inline src/components/courses.rs:116]: Just a heads up, this `gap-2` provides some jank on desktop when previewing schedules. Not a showstopper but worth noting.

[Screencast_20241202_203713.webm](https://github.com/user-attachments/assets/ebdb3302-495d-4ba1-ba4a-692940c576ba)

**Review by malcolmseyd — APPROVED** [review]
Looks good to me, I'm super impressed with how it looks right now! I left a single comment regarding a small UI jank, but I'd say this is good enough to merge.

Leo LGTM

File: src/components/courses.rs (hunk 1 above)
Comment (brennanmcmicking) [reply src/components/courses.rs:116]: Yeah I'm still not sure how to handle the preview. We might need to disable it for now and send it back to the drawing board. I tried putting a delay on the hover but that just makes it janky in a new and novel way.

Comment (malcolmseyd) [reply src/components/courses.rs:116]: The only way I can think of is removing the gap. We would then have to separate items using border or outline or something, or add some custom JS that retains the last hover state when still inside the parent div.

**Pull request merged**